### ISSUE TITLE
False positive for PLE0118 when mixing globals and f-strings

### ISSUE THREAD

**Issue by PhilReinhold** [charliermarsh/ruff#1309]
It seems like we get a false positive from using globals inside of f-strings

```python
# test.py
x = 1

def f():
    global x
    x = x + 1
    print(f"{x=}")
```

Then

```
$ ruff --version
ruff 0.0.189

$ ruff --select PLE test.py --show-source
Found 1 error(s).
test.py:7:11: PLE0118 Name `x` is used prior to global declaration on line 5
  |
7 |     print(f"{x=}")
  |           ^^^^^^^ PLE0118
  |
```

**Comment by charliermarsh** [comment]
Ah yeah, this makes sense. Will fix.

(We have to treat f-strings in a special way everywhere, because RustPython doesn't properly adjust the locations of the nested expressions.)
